TombLib/TombLib/Catalogs/TEN Node Catalogs/Particles.lua

```diff
@@ -0,0 +1,114 @@
+--!Arguments "NewLine, Color, 50, Start color", "Color, 50, End color"
+--!Arguments "NewLine, Enumeration, 100, [Opaque | Alpha test | Add | No Z test | Subtract | Wireframe | Exclude | Screen | Lighten | Alpha blend], Blending method for particles"
+--!Arguments "NewLine, Numerical, 33, [ -32000 | 32000 | 0 ], Start size." "Numerical, 33, [ -32000 | 32000 | 0 ], End size." "Numerical, 33, [ 0 | 32000 | 0 ], Lifetime (in seconds)"
+--!Arguments "NewLine, Boolean, 50, Damage" , "Boolean, 50, Poison"
+
+	LevelFuncs.Engine.Node.StaticParticles=function(entity, velocity, spriteID, gravity, rotation, startColor, endColor, blendID, startSize, endSize, life, damage, poison)
+
+		local origin = TEN.Objects.GetStaticByName(entity):GetPosition()
+		local blendmode = LevelFuncs.Engine.Node.GetBlendMode(blendID)
+
+		TEN.Effects.EmitParticle(origin, velocity, spriteID, gravity, rotation, startColor, endColor, blendmode, startSize, endSize, life, damage, poison)
+		
+	end
+
+--!Name "Emit lightning arc"
+--!Section "Particles"
+--!Conditional "False"
+--!Description "Emit a lightning arc between two points in 3D space"
+--!Arguments "Newline, Moveables, 50, Source position", "Moveables, 50, Destination position"
+--!Arguments "NewLine, Color, 100, Color of Lightning Effect"
+--!Arguments "Newline, Number, 25, [ 0 | 4.2 | 1 ], Lifetime in seconds." , "Number, 25, [ 1 | 255 | 0 ], Effect strength", "Number, 25, [ 1 | 127 | 0 ], Beam width", "Number, 25, [ 1 | 127 | 0 ], Detail level\n 1 is 1 segment between the two points for example."
+--!Arguments "Newline, Boolean, 50, Toggle smooth effect.", "Boolean, 50, Toggle end drift."
+--!Arguments "NewLine, Boolean, 50, Add source light" , "Boolean, 50, Add destination light"
+
+LevelFuncs.Engine.Node.LightningArc = function(source, dest, color, lifetime, amplitude, beamWidth, detail, smooth, endDrift, sourcelight, destlight)
+	
+	local randomiserX = math.random(-64,64)
+	local randomiserZ = math.random(-256,256)
+	
+    local entity = TEN.Objects.GetMoveableByName(source)
+	
+    local startingpoint = entity:GetPosition()
+		startingpoint.x = startingpoint.x - randomiserX
+		startingpoint.y = startingpoint.y - 64
+		startingpoint.z = startingpoint.z - randomiserZ 	
+		
+	local endingpoint = TEN.Objects.GetMoveableByName(dest):GetPosition()
+		endingpoint.x = endingpoint.x - randomiserX
+		endingpoint.y = endingpoint.y - 64
+		endingpoint.z = endingpoint.z - randomiserZ 
+		
+	local beamrandom = math.random((beamWidth-10),(beamWidth+10))
+	local amprandom =  math.random((amplitude-10),(amplitude+10))
+	
+	local sourceLightCheck 
+		if sourcelight == true
+		then
+			TEN.Effects.EmitLight(startingpoint, color, math.random(1,10))
+		end
+
+	local destLightCheck 
```

Comment:
What is the purpose of this line? What is `destLightCheck`? It is never assigned and never used.

Reply:
As with above. I am not sure if I can just have:

```lua

if (sourcelight == true) then
TEN.Effects.EmitLight(startingpoint, color, math.random( 1, 10 ))
		end

if (destlight == true) then 
			TEN.Effects.EmitLight(endingpoint, color, math.random( 1, 10 ))
		end
		
```

instead of:

```lua

local sourceLightCheck 
		if (sourcelight == true) then
			TEN.Effects.EmitLight(startingpoint, color, math.random( 1, 10 ))
		end

	local destLightCheck 
		if (destlight == true) then 
			TEN.Effects.EmitLight(endingpoint, color, math.random( 1, 10 ))
		end
		
		```

Reply:
Again, I don't understand what are you trying to do. Where did you get the concept that you need `local whateverName` line before your conditional statement?